typewise-alert.h

```diff
@@ -1,32 +1,35 @@
-#pragma once
+#ifndef TYPEWISEALERT
+#define TYPEWISEALERT
 
-typedef enum {
-  PASSIVE_COOLING,
-  HI_ACTIVE_COOLING,
-  MED_ACTIVE_COOLING
-} CoolingType;
+#define PASSIVE_COOLING_LOWER_LIMIT 0
+#define PASSIVE_COOLING_UPPER_LIMIT 35
+#define HI_ACTIVE_COOLING_LOWER_LIMIT 0
+#define HI_ACTIVE_COOLING_UPPER_LIMIT 45
+#define MED_ACTIVE_COOLING_LOWER_LIMIT 0
+#define MED_ACTIVE_COOLING_UPPER_LIMIT 40
 
-typedef enum {
-  NORMAL,
-  TOO_LOW,
-  TOO_HIGH
-} BreachType;
+enum BreachType {NORMAL, TOO_LOW, TOO_HIGH};
+enum AlertAcknowledgement {CONTROLLER_ALERT, NORMAL_EMAIL_ALERT, TOO_LOW_EMAIL_ALERT, TOO_HIGH_EMAIL_ALERT};
 
-BreachType inferBreach(double value, double lowerLimit, double upperLimit);
-BreachType classifyTemperatureBreach(CoolingType coolingType, double temperatureInC);
+#include <iostream>
+#include <map>
+#include <assert.h>
+#include "AlertTarget.h"
+#include "CoolingType.h"
 
-typedef enum {
-  TO_CONTROLLER,
-  TO_EMAIL
-} AlertTarget;
-
-typedef struct {
-  CoolingType coolingType;
-  char brand[48];
-} BatteryCharacter;
-
-void checkAndAlert(
-  AlertTarget alertTarget, BatteryCharacter batteryChar, double temperatureInC);
-
-void sendToController(BreachType breachType);
-void sendToEmail(BreachType breachType);
+class TypewiseAlert
+{
+    AlertTarget *alertTarget;
+    CoolingType *coolingType;
+    BreachType breachType;
+    
+    public:
+    TypewiseAlert(AlertTarget *alertTarget, CoolingType *coolingType, double temperatureInCelsius);
+    AlertTarget* getAlertTargetInstance();
+    void setBreachType(BreachType breachType);
```

Comment:
set breach type is a public method, whereas as per code i see that it is required to be set only via member methods.
Please correct if it is indeed the case.

Reply:
Changed setBreachType() method as a private method.